internal/discord/commands.go

```diff
@@ -10,7 +12,75 @@ import (
 //	user: 	User
 //	reason: string
 func (d *Discord) Kick(s *discordgo.Session, i *discordgo.InteractionCreate) {
-	return
+	options := i.ApplicationCommandData().Options
+	optionMap := make(map[string]*discordgo.ApplicationCommandInteractionDataOption, len(options))
+	for _, opt := range options {
+		optionMap[opt.Name] = opt
+	}
+
+	userToKick := optionMap["user"].UserValue(nil).ID
+
+	// Check if user exists in guild
+	err := d.CheckUserInGuild(i.GuildID, userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		StartInteraction(s, i.Interaction, tempstr)
+		return
+	}
+
+	// DM the user regarding the kick
+	channel, err := s.UserChannelCreate(userToKick)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not send a DM to user %v", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		// Abort kick if failed to DM?
+		//return
+		StartInteraction(s, i.Interaction, tempstr)
+	} else {
+		// Get guild name
+		var guildname string
+		guild, err := d.Session.Guild(i.GuildID)
+		if err != nil {
+			fmt.Printf("Unable to find guild name: %v", err)
+			guildname = i.GuildID
+		} else {
+			guildname = guild.Name
+		}
+
+		tempstr := fmt.Sprintf("You are being kicked from %v for the reason:\n%v",
+			guildname,
+			optionMap["reason"].StringValue(),
+		)
+
+		_, err = s.ChannelMessageSend(channel.ID, tempstr)
+		if err != nil {
+			tempstr := fmt.Sprintf("Could not send a DM to user %v", userToKick)
+			fmt.Printf("%v: %v\n", tempstr, err)
+			// Abort kick if failed to DM?
+			//return
+			StartInteraction(s, i.Interaction, tempstr)
+		}
+	}
+
+	// Attempt to kick user
+	if len(optionMap["reason"].StringValue()) > 0 {
+		err = d.Session.GuildMemberDeleteWithReason(i.GuildID, userToKick, optionMap["reason"].StringValue())
+	} else {
+		StartInteraction(s, i.Interaction, "Please provide a reason for the kick.")
+		return
+	}
+
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not kick user <@%v>", userToKick)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		StartInteraction(s, i.Interaction, tempstr)
+	} else {
+		tempstr := fmt.Sprintf("User <@%v> has been kicked.", userToKick)
+		fmt.Printf("%v\n", tempstr)
```

Comment:
let's also make sure that we are logging the reason for the kick too. in the future we'll need to make an embed formatting which mod kicked who and why